src/Transition/CreateNewIDFUsingRulesV24_2_0.f90

```diff
@@ -280,7 +298,24 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
 
 ! Do any kind of Preprocessing that is needed here (eg: a first pass on objects to store some attributes etc)
 
+          DO i = 1, size(SPMTypes)
+              TotSPMs = TotSPMs + GetNumObjectsFound(SPMTypes(i))
+          ENDDO
+          print *, "Found ", TotSPMs, " SPMs"
+          IF(ALLOCATED(SPMNames)) DEALLOCATE(SPMNames)
+          ALLOCATE(SPMNames(TotSPMs))
+          DO i = 1, size(SPMTypes)
+              DO spmNum=1,GetNumObjectsFound(SPMTypes(i))
+                  CALL GetObjectItem(SPMTypes(i),spmNum,Alphas,NumAlphas,Numbers,NumNumbers,Status)
+                  ! VerifyName(TRIM(Alphas(1)), SPMNames, spmIndex, SPMErrorFound, IsBlank, "SetpointManager Unicity");
+                  IF (FindItemInList(TRIM(Alphas(1)), SPMNames, spmIndex) /= 0) THEN
+                      CALL ShowFatalError('SetpointManager Unicity of Names: SPM of type '//TRIM(SPMTypes(i))//' has a name already found='//TRIM(Alphas(1)),Auditf)
```

Comment:
Could you just append a loop or counter index to make the name unique and continue searching?